snf-cyclades-app/synnefo/logic/servers.py

```diff
@@ -172,9 +173,13 @@ def create(userid, name, password, flavor, image_id, metadata={},
             meta_value=val,
             vm=vm)
 
+    if key_name is not None:
```

Comment:
What if `key_name` is None? `public_key` will not be defined but `create_server()` will still try to use it. This will create a `UnboundLocalError` exception.